doseCalc/+DoseEngines/matRad_PencilBeamEngineAbstract.m

```diff
@@ -0,0 +1,630 @@
+            xCoordsV       = this.xCoordsV_vox(:)*dij.ctGrid.resolution.x-currBeam.isoCenter(1);
+            yCoordsV       = this.yCoordsV_vox(:)*dij.ctGrid.resolution.y-currBeam.isoCenter(2);
+            zCoordsV       = this.zCoordsV_vox(:)*dij.ctGrid.resolution.z-currBeam.isoCenter(3);
+            coordsV        = [xCoordsV yCoordsV zCoordsV];
+
+            xCoordsVdoseGrid = this.xCoordsV_voxDoseGrid(:)*dij.doseGrid.resolution.x-currBeam.isoCenter(1);
+            yCoordsVdoseGrid = this.yCoordsV_voxDoseGrid(:)*dij.doseGrid.resolution.y-currBeam.isoCenter(2);
+            zCoordsVdoseGrid = this.zCoordsV_voxDoseGrid(:)*dij.doseGrid.resolution.z-currBeam.isoCenter(3);
+            coordsVdoseGrid  = [xCoordsVdoseGrid yCoordsVdoseGrid zCoordsVdoseGrid];
+
+            % Get Rotation Matrix
+            % Do not transpose matrix since we usage of row vectors &
+            % transformation of the coordinate system need double transpose
+
+            currBeam.rotMat_system_T = matRad_getRotationMatrix(currBeam.gantryAngle,currBeam.couchAngle);
+
+            % Rotate coordinates (1st couch around Y axis, 2nd gantry movement)
+            rot_coordsV         = coordsV*currBeam.rotMat_system_T;
+            rot_coordsVdoseGrid = coordsVdoseGrid*currBeam.rotMat_system_T;
+            
+            rot_coordsV = rot_coordsV - currBeam.sourcePoint_bev;
+            rot_coordsVdoseGrid = rot_coordsVdoseGrid - currBeam.sourcePoint_bev;
+
+            % calculate geometric distances
+            geoDistVdoseGrid{1}= sqrt(sum(rot_coordsVdoseGrid.^2,2));
+
+            % Calculate radiological depth cube
+            matRad_cfg.dispInfo('matRad: calculate radiological depth cube... ');
+
+            ct.cube = this.cubeWED;
+            if this.keepRadDepthCubes
+                [radDepthVctGrid, currBeam.radDepthCube] = matRad_rayTracing(currBeam,ct,this.VctGrid,rot_coordsV,this.effectiveLateralCutOff);
+                currBeam.radDepthCube{1} = matRad_interp3(dij.ctGrid.x,  dij.ctGrid.y,   dij.ctGrid.z, currBeam.radDepthCube{1}, ...
+                    dij.doseGrid.x,dij.doseGrid.y',dij.doseGrid.z,'nearest');
+                this.radDepthCubes{i} = currBeam.radDepthCube{1};
+            else
+                radDepthVctGrid = matRad_rayTracing(currBeam,ct,this.VctGrid,rot_coordsV,this.effectiveLateralCutOff);
+            end
+            
+            % interpolate radiological depth cube to dose grid resolution
+            radDepthVdoseGrid = this.interpRadDepth(ct,1:numel(ct.numOfCtScen),this.VctGrid,this.VdoseGrid,dij.ctGrid,dij.doseGrid,radDepthVctGrid);
+            
+            % limit rotated coordinates to positions where ray tracing is availabe
+            %radDepthsMat = cellfun(@(radDepthCube) matRad_interp3(dij.ctGrid.x,  dij.ctGrid.y,   dij.ctGrid.z,radDepthCube,dij.doseGrid.x,dij.doseGrid.y',dij.doseGrid.z,'nearest'),radDepthsMat,'UniformOutput',false);
+            
+            coordIsValid = ~isnan(radDepthVdoseGrid{1});
```

Comment:
the rad depth are not considered correctly here, ithink for 4D it only take the raddepth from the 1 ct scen, not sure right now what the best fix is